### PR TITLE
use std::unique_ptr instead of std::auto_ptr

### PR DESCRIPTION
**Pull request by ClausKlein**
prevent include of ace/Auto_Ptr.h for TAO at least

related https://github.com/DOCGroup/ACE_TAO/issues/1301

### CODE REVIEW

File: TAO/tests/Bug_3542_Regression/DllOrb.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ class bug_3542_Export DllOrb: public ACE_Task_Base
   virtual int svc ();
 
 private:
-  auto_ptr < ACE_Thread_Barrier >      ma_barrier_;
+  std::unique_ptr < ACE_Thread_Barrier >    ma_barrier_;
```
Comment (jwillemsen) [inline TAO/tests/Bug_3542_Regression/DllOrb.h:28]: No need for spaces between `<> `and `ACE_Thread_Barrier` (only `<::` is problematic with some compilers, we always use `< ::` in that case), align `ma_barrier_` with the other members

File: TAO/tests/Bug_3251_Regression/DllOrb.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ class bug_3251_Export DllOrb: public ACE_Task_Base
   virtual int svc ();
 
 private:
-  auto_ptr < ACE_Thread_Barrier >      ma_barrier_;
+  std::unique_ptr < ACE_Thread_Barrier >    ma_barrier_;
```
Comment (jwillemsen) [inline TAO/tests/Bug_3251_Regression/DllOrb.h:28]: No need for spaces between `<> `and `ACE_Thread_Barrier` (only `<::` is problematic with some compilers, we always use `< ::` in that case), align `ma_barrier_` with the other members

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h [hunk 3]
```diff
@@ -28,7 +28,7 @@ class Server_Task : public ACE_Task_Base
   ACE_ARGV my_args (args_.c_str());
 
   // Initialize Server ORB in new thread
-  ACE_auto_ptr_reset (server_, new Server (my_args.argc(), my_args.argv()));
+  server_ = std::make_unique<Server>(my_args.argc(), my_args.argv());
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/Server_Task.h:31]: `std::make_unique` is a C++14 feature so you can't use it as much as we would like to do that.

File: TAO/tests/Compression/RLECompressorTest.cpp [hunk 4]
```diff
@@ -9,10 +9,9 @@
 
 #include <memory>
 
-// Older versions of GCC do not support std::unique_ptr!
-struct ACE_Byte_Array_ptr : std::auto_ptr<ACE_Byte> {
+struct ACE_Byte_Array_ptr : std::unique_ptr<ACE_Byte> {
```
Comment (jwillemsen) [inline TAO/tests/Compression/RLECompressorTest.cpp:12]: This struct looks to be something we should completely remove and use a different construct in the code that uses this struct

**Review by jwillemsen — CHANGES_REQUESTED** [review]

File: TAO/tests/Bug_3542_Regression/DllOrb.h (hunk 1 above)
Comment (ClausKlein) [reply TAO/tests/Bug_3542_Regression/DllOrb.h:28]: Would you like to have a .clang-format to enforce our code style?

File: TAO/tests/Bug_3251_Regression/DllOrb.h (hunk 2 above)
Comment (ClausKlein) [reply TAO/tests/Bug_3251_Regression/DllOrb.h:28]: A `.clang-format` file would be helpfully

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 3 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h:31]: You are right, but it builds on CI?

File: TAO/tests/Compression/RLECompressorTest.cpp (hunk 4 above)
Comment (ClausKlein) [reply TAO/tests/Compression/RLECompressorTest.cpp:12]: I do not know this code, it was not my intention to refactor ACE_TAO tests at all.

File: TAO/tests/Bug_3542_Regression/DllOrb.h (hunk 1 above)
Comment (jwillemsen) [reply TAO/tests/Bug_3542_Regression/DllOrb.h:28]: Feel free to add one in a different PR

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 3 above)
Comment (jwillemsen) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h:31]: CI doesn't build all, a full build probably takes up to 10 hours

File: TAO/tests/Bug_3542_Regression/DllOrb.h (hunk 1 above)
Comment (ClausKlein) [reply TAO/tests/Bug_3542_Regression/DllOrb.h:28]: OK

File: TAO/tests/Bug_3251_Regression/DllOrb.h (hunk 2 above)
Comment (ClausKlein) [reply TAO/tests/Bug_3251_Regression/DllOrb.h:28]: OK

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 3 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h]: OK

File: TAO/tests/Compression/RLECompressorTest.cpp (hunk 4 above)
Comment (jwillemsen) [reply TAO/tests/Compression/RLECompressorTest.cpp:12]: At least this should be a new issue, this is a workaround for an older compiler which should be removed

File: TAO/tests/Oneway_Send_Timeouts/ORB_Task.h [hunk 5]
```diff
@@ -7,22 +7,22 @@
...
+public:
   ORB_Task (CORBA::ORB_ptr orb)
     : orb_ (CORBA::ORB::_duplicate (orb))
-    {
-      if (CORBA::is_nil (orb_.in()) == 1) {
-  ACE_ERROR ((LM_ERROR, "ORB_Task> Ctr> Orb is NULL\n"));
-      }
-    };
+  {
+    if (CORBA::is_nil (orb_.in ()) == 1) {
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: Normal ACE coding style would be `{` on the next line, but we have so many older tests contributed by a lot of people that coding styles are different

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h [hunk 6]
```diff
@@ -3,34 +3,32 @@
...
+  ~Server_Task () { this->force_shutdown (); }
 
   virtual int svc ()
   {
     bool initializer = false;
     {
       ACE_GUARD_RETURN (TAO_SYNCH_MUTEX, ace_mon, this->mutex_, -1);
-      if (server_.get() == 0) {
-  ACE_ARGV my_args (args_.c_str());
+      if (server_.get () == 0) {
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/Server_Task.h:24]: No need to call `get ()` when this is now a `unique_ptr`, there is a `operator bool`

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h [hunk 7]
```diff
@@ -3,34 +3,32 @@
...
+        ACE_ARGV my_args (args_.c_str ());
 
-  // Initialize Server ORB in new thread
-  ACE_auto_ptr_reset (server_, new Server (my_args.argc(), my_args.argv()));
-  ACE_ASSERT (server_.get() != 0);
-  initializer = true;
+        // Initialize Server ORB in new thread
+        // server_ = std::make_unique<Server>(my_args.argc(), my_args.argv());
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/Server_Task.h:28]: You could check `ACE_HAS_CPP14` and use at that moment `make_unique`, if not set, use new. That way we can easily modernize the code at the moment we migrate to C++14

File: TAO/tests/Oneway_Send_Timeouts/ORB_Task.h (hunk 5 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: No Problem, can be changed

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 7 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h:28]: I do not understand why here 2 args are used?

```c++
server_ = std::make_unique<Server>(my_args);
```
Should work, or not?

File: TAO/tests/Oneway_Send_Timeouts/ORB_Task.h (hunk 5 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: @jwillemsen Q: what should be the value for `ColumnLimit: 79` ?

Comment (jwillemsen) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: That is very small, would go for at least 120, maybe even 160

Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: @jwillemsen Q: and what is about this? `AlwaysBreakAfterReturnType: None`

Comment (jwillemsen) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: Does https://htmlpreview.github.io/?https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/docs/ACE-guidelines.html help?

Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h:17]: **Yes**! But **No** in respect to `AlwaysBreakAfterReturnType: None` the examples are inconsistent!

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 6 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h:24]: OK

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 7 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h:28]: OK

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h [hunk 8]
```diff
@@ -47,7 +53,8 @@ class Server_Task : public ACE_Task_Base
 
   bool ready ()
   {
-    if (server_.get() != 0) {
+    if (server_.get () != 0)
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/Server_Task.h:56]: get() can be removed here, we can use operator bool now, more places in this file

File: TAO/tests/Oneway_Send_Timeouts/ORB_Task.h (hunk 5 above)
Comment (jwillemsen) [reply TAO/tests/Oneway_Send_Timeouts/ORB_Task.h]: I think we should use `RTBS_All`

File: TAO/tests/Oneway_Send_Timeouts/main.cpp [hunk 9]
```diff
@@ -126,7 +140,8 @@ MyMain::run ()
 
 MyMain::~MyMain ()
 {
-  if (!shutdown_) {
+  if (! shutdown_)
```
Comment (jwillemsen) [inline TAO/tests/Oneway_Send_Timeouts/main.cpp:143]: We normally don't use a space between ! and shutdown_

File: TAO/tests/Bug_3251_Regression/DllOrb.h [hunk 10]
```diff
@@ -1,34 +1,34 @@
...
   virtual int fini ();
 
   virtual int svc ();
 
 private:
-  auto_ptr < ACE_Thread_Barrier >      ma_barrier_;
-  CORBA::ORB_var                       mv_orb_;
-  PortableServer::POA_var              mv_rootPOA_;
-  PortableServer::POAManager_var       mv_poaManager_;
+  std::unique_ptr<ACE_Thread_Barrier> ma_barrier_;
```
Comment (jwillemsen) [inline TAO/tests/Bug_3251_Regression/DllOrb.h:28]: In general we do not align the members, these test had that but now we define a generic clang-format for ACE/TAO I would prefer to update these tests to follow the guidelines

File: TAO/tests/Oneway_Send_Timeouts/main.cpp (hunk 9 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/main.cpp:143]: I found it very often

File: TAO/tests/Bug_3251_Regression/DllOrb.h (hunk 10 above)
Comment (ClausKlein) [reply TAO/tests/Bug_3251_Regression/DllOrb.h:28]: OK, I will change it

File: TAO/tests/Oneway_Send_Timeouts/main.cpp (hunk 9 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/main.cpp]: ok, done

File: TAO/tests/Oneway_Send_Timeouts/Server_Task.h (hunk 8 above)
Comment (ClausKlein) [reply TAO/tests/Oneway_Send_Timeouts/Server_Task.h]: done

File: TAO/tests/Compression/RLECompressorTest.cpp (hunk 4 above)
Comment (ClausKlein) [reply TAO/tests/Compression/RLECompressorTest.cpp:12]: Can you please open a Ticket, I am on Holiday next 2 weeks.

**Review by jwillemsen — APPROVED** [review]

**Comment by jwillemsen** [comment]
Did you compile the tests after your changes, they didn't compile, see #1542

**Comment by jwillemsen** [comment]
And #1543, hopefully the last one

**Pull request merged**